salt/config/__init__.py

```diff
@@ -2560,8 +2560,6 @@ def is_profile_configured(opts, provider, profile_name):
 
     if driver not in non_image_drivers:
         required_keys.append('image')
-    elif driver == 'vmware':
-        required_keys.append('clonefrom')
```

Comment:
Removing the check totally is a bad idea. Either ``image`` or ``cloneform`` should be specified for the VMware driver. Renaming ``guest_id`` option to ``image`` would be the way to go.